src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -952,14 +951,28 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
                         for (i = 0; (uint)i < maxIters; i++)
                         {
                             utf16Data = Unsafe.ReadUnaligned<Vector128<short>>(pInputBuffer);
-                            if (!Sse41.TestZ(utf16Data, nonAsciiUtf16DataMask))
+
+                            if (AdvSimd.IsSupported)
                             {
-                                goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
-                            }
+                                if (AdvSimd.CompareTest(utf16Data, nonAsciiUtf16DataMask).ToScalar() == 0)
+                                {
+                                    goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
+                                }
 
-                            // narrow and write
+                                Vector64<sbyte> lower = AdvSimd.ExtractNarrowingSaturateLower(utf16Data);
+                                Vector128<sbyte> source = AdvSimd.ExtractNarrowingSaturateUpper(lower, AdvSimd.LoadVector128((short*)pInputBuffer));
+                                AdvSimd.Store((ulong*)pOutputBuffer, source.AsUInt64());
```

Comment:
This is writing more than the x86 version. I'm also a bit surprised we aren't exposing `AdvSimd.StoreScalar`, as it looks like it would map to `LDR`/`STR`. @echesakovMSFT, did we miss getting an issue up for these variants?

Reply:
Looks like we have `StoreSelectedScalar`, which can be used as a stop gap.

Reply:
@tannergooding I think we have all the methods corresponding STR operating on `Vector64<ulong>` and similar - we just named them as `AdvSimd.Store` not `AdvSimd.StoreScalar`